internal/service/memory/helper_service.go

```diff
@@ -47,63 +53,90 @@ func (s *Service) HandleCookie(cookieValue string) (*http.Cookie, string, error)
 		Name: usr,
 	}
 
-	if s.repo.GetShu().Settings.TypeStorage == config.DBStorage {
-		if cookieValue == "" {
-			usr = "user" + strconv.Itoa(s.GetCounter())
-			u.Name = usr
-			if encoded, err := s.secure.Encode("user", u); err == nil {
-				cookie = &http.Cookie{
-					Name:  "user",
-					Value: encoded,
-				}
-				return cookie, usr, nil
-			} else {
-				return nil, "", err
+	if cookieValue == "" {
+		usr = "user" + strconv.Itoa(s.GetCounter())
+		u.Name = usr
+		if encoded, err := s.secure.Encode("user", u); err == nil {
+			cookie = &http.Cookie{
+				Name:  "user",
+				Value: encoded,
 			}
-		}
-		if err := s.secure.Decode("user", cookieValue, u); err != nil {
+			return cookie, usr, nil
+		} else {
 			return nil, "", err
 		}
-		fmt.Println("u: ", u)
-		if _, ok := s.repo.GetShu().MapUser[u.Name]; ok {
-			return nil, u.Name, nil
-		} else {
-			usr = "user" + strconv.Itoa(s.GetCounter())
-			u.Name = usr
-			if encoded, err := s.secure.Encode("user", u); err == nil {
-				cookie = &http.Cookie{
-					Name:  "user",
-					Value: encoded,
-				}
-				s.repo.GetShu().MapUser[usr] = true
-				return cookie, usr, nil
-			} else {
-				return nil, "", err
+	}
+	if err := s.secure.Decode("user", cookieValue, u); err != nil {
+		return nil, "", err
+	}
+	s.mu.RLock()
+	_, ok := s.shu.MapUser[u.Name]
+	s.mu.RUnlock()
+	if ok {
+		return nil, u.Name, nil
+	} else {
+		usr = "user" + strconv.Itoa(s.GetCounter())
+		u.Name = usr
+		if encoded, err := s.secure.Encode("user", u); err == nil {
+			cookie = &http.Cookie{
+				Name:  "user",
+				Value: encoded,
 			}
+			s.mu.Lock()
+			s.shu.MapUser[usr] = true
+			s.mu.Unlock()
+			return cookie, usr, nil
+		} else {
+			return nil, "", err
 		}
-	} else {
-		return nil, "", nil
 	}
 }
 
 // CreateMapUser создает Map пользователей
-func CreateMapUser(shu *ShortURLAttr) (map[string]bool, error) {
-	mapUser, err := dbstorage.CreateMapUser(shu.DB)
+func CreateMapUser(ctx context.Context, shu *ShortURLAttr) (map[string]bool, error) {
+	mapUser, err := dbstorage.CreateMapUser(ctx, shu.DB)
 	if err != nil {
 		return nil, err
 	}
 	shu.Cntr = len(mapUser)
 	return mapUser, nil
 }
 
+func (s *Service) setDelMemory(arrShortURL []string, usr string) error {
+	var err error
+
+	for _, shortURL := range arrShortURL {
+		intShortURL, err := strconv.Atoi(shortURL)
+		if err != nil {
+			return err
+		}
+		if val, ok := s.shu.MapURL[intShortURL]; ok {
+			if val.Usr == usr {
+				s.mu.Lock()
+				mapURLVal := s.shu.MapURL[intShortURL]
+				mapURLVal.IsDeleted = "true"
+				s.shu.MapURL[intShortURL] = mapURLVal
+				s.mu.Unlock()
+			}
+		}
+	}
+	if s.shu.Settings.TypeStorage == config.FileStorage {
+		s.changeFileStorage()
+	}
+	return err
+}
+
 // DelURLs делает отметку об удалении коротких URL-ы определенного пользователя
 func (s *Service) DelURLs(cookieValue string, buf bytes.Buffer) (bool, error) {
```

Comment:
```suggestion
func (s *Service) DelURLs(cookieValue string, buf bytes.Buffer) error {
```
Насколько я успел заметить, булево значение не используется. Нет смысла его оставлять в выводе функции.

Reply:
Действительно, bool значение теперь не используется, убрал.